### PR TITLE
Use LazyListUpdateProcessor for Android views

### LINKED ISSUE #1552: Migrate ViewLazyList to use LazyListUpdateProcessor
LazyListUpdateProcessor is a fork of WindowedLazyList first shipped on iOS. We should migrate everything to that fork and delete the old class.

### PR DESCRIPTION
**Pull request by swankjesse**
Closes: https://github.com/cashapp/redwood/issues/1552

### CODE REVIEW

File: redwood-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/lazylayout/uiview/UIViewLazyList.kt [hunk 1]
```diff
@@ -250,15 +250,13 @@ internal class LazyListContainerCell(
 
     // Unbind the cell when its view is detached from the table.
     if (superview != null && newSuperview == null) {
-      removeAllSubviews()
```
Comment (swankjesse) [inline redwood-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/lazylayout/uiview/UIViewLazyList.kt:257]: This is now performed by `unbind()` below.

File: redwood-lazylayout-view/src/main/kotlin/app/cash/redwood/lazylayout/view/ViewLazyList.kt [hunk 2]
```diff
@@ -184,71 +200,46 @@ internal open class ViewLazyList private constructor(
...
-          if (holder.container.childCount == 0) {
-            val placeholder = placeholders.takeOrNull()
-            if (placeholder != null) {
-              placeholder.value.layoutParams = layoutParams
-              holder.container.addView(placeholder.value)
-              holder.itemView.updateLayoutParams { height = WRAP_CONTENT }
-            } else if (holder.container.height == 0) {
-              // This occurs when the ViewHolder has been freshly created, so we set the container
-              // to a non-zero height so that it's visible.
-              holder.itemView.updateLayoutParams { height = lastItemHeight }
```
Comment (swankjesse) [inline redwood-lazylayout-view/src/main/kotlin/app/cash/redwood/lazylayout/view/ViewLazyList.kt:232]: Big regression here . . . I need to add a hook to create placeholders on-demand.

Comment (swankjesse) [reply redwood-lazylayout-view/src/main/kotlin/app/cash/redwood/lazylayout/view/ViewLazyList.kt:232]: Tracking here: https://github.com/cashapp/redwood/issues/1571

File: redwood-lazylayout-compose/src/commonTest/kotlin/app/cash/redwood/lazylayout/compose/LazyListTest.kt [hunk 3]
```diff
@@ -84,188 +70,44 @@ class LazyListTest {
...
   }
-
-private class RedwoodLazyLayoutFakeWidgetFactory(
-  private val windowedLazyList: WindowedLazyList<WidgetValue>,
-) : RedwoodLazyLayoutWidgetFactory<WidgetValue> {
-  override fun LazyList() = windowedLazyList
-  override fun RefreshableLazyList() = error("unexpected call")
 }
-
-private class FakeWindowedLazyList : WindowedLazyList<WidgetValue>(NoOpListUpdateCallback) {
```
Comment (swankjesse) [inline redwood-lazylayout-compose/src/commonTest/kotlin/app/cash/redwood/lazylayout/compose/LazyListTest.kt:166]: Previously we used the compose-side LazyList to drive this on the widget-side.

With FakeProcessor we’ve got a test facet for the widget side and I don’t think this integration test is necessary.

File: redwood-lazylayout-view/src/main/kotlin/app/cash/redwood/lazylayout/view/ViewLazyList.kt [hunk 4]
```diff
@@ -151,24 +153,38 @@ internal open class ViewLazyList private constructor(
...
-  @SuppressLint("NotifyDataSetChanged")
   override fun itemsBefore(itemsBefore: Int) {
-    items.itemsBefore = itemsBefore
-
-    // TODO Replace notifyDataSetChanged with atomic change events
-    //  notifyItemRangeInserted causes an onScrolled event to be emitted.
-    //  This incorrectly updates the viewport, which then shifts the loaded items window.
-    //  This then increases the value of itemsBefore,
-    //  and the cycle continues until the backing dataset is exhausted.
-    adapter.notifyDataSetChanged()
```
Comment (swankjesse) [inline redwood-lazylayout-view/src/main/kotlin/app/cash/redwood/lazylayout/view/ViewLazyList.kt:163]: # 🎉

**Comment by swankjesse** [comment]
https://github.com/cashapp/redwood/assets/133019/bcbb54df-622e-4b99-9f05-40e55472474d

**Review by JakeWharton — APPROVED** [review]
Can't say I fully understood everything, but LGTM

**Pull request merged**